src/simple_http.rs

```diff
@@ -73,29 +75,50 @@ impl SimpleHttpTransport {
     }
 
     fn request<R>(&self, req: impl serde::Serialize) -> Result<R, Error>
+    where
+        R: for<'a> serde::de::Deserialize<'a>,
+    {
+        match self.try_request(req) {
+            Ok(response) => Ok(response),
+            Err(err) => {
+                *self.sock.lock().unwrap() = None;
```

Comment:
I think we should add a mutex poisoning error to `Error` and then use `?` rather than unwrapping. In theory the user might want to catch this and recover somehow.

Reply:
The mutex can only be poisoned if `try_request` panics. I think since we are doing proper error handling in that function it should never happen. But I'm not 100% sure on that.

Reply:
I considered this, and I'm not 100% either. If you like you can keep the `.unwrap` and just add a comment saying that we're pretty confident that we'll never panic when the mutex is open.